Todo/TasksViewController.swift

```diff
@@ -0,0 +1,353 @@
+                self.deleteTask(doc)
+            }
+            delete.backgroundColor = UIColor(red: 1.0, green: 0.23, blue: 0.19, alpha: 1.0)
+            
+            let update = UITableViewRowAction(style: .Normal, title: "Edit") {
+                (action, indexPath) -> Void in
+                // Dismiss row actions:
+                tableView.setEditing(false, animated: true)
+                // Display update list dialog:
+                let document = self.taskRows![indexPath.row].document!
+                
+                Ui.showTextInputDialog(
+                    onController: self,
+                    withTitle: "Edit Task",
+                    withMessage:  nil,
+                    withTextFieldConfig: { (textField) -> Void in
+                        textField.placeholder = "Task"
+                        textField.text = document["task"] as? String
+                        textField.autocapitalizationType = .Sentences
+                    },
+                    onOk: { (task) -> Void in
+                        self.updateTask(document, withTitle: task)
+                    }
+                )
+            }
+            update.backgroundColor = UIColor(red: 0.0, green: 0.48, blue: 1.0, alpha: 1.0)
+            
+            return [delete, update]
+    }
+    
+    // MARK: - UISearchController
+    
+    func updateSearchResultsForSearchController(searchController: UISearchController) {
+        let text = searchController.searchBar.text ?? ""
+        let active = !text.isEmpty
+        tasksLiveQuery.postFilter = active ? NSPredicate(format: "key2 CONTAINS[cd] %@", text) : nil
+        tasksLiveQuery.queryOptionsChanged()
+    }
+    
+    // MARK: - UIImagePickerControllerDelegate
+    
+    func imagePickerController(picker: UIImagePickerController,
+        didFinishPickingImage image: UIImage, editingInfo: [String : AnyObject]?) {
+            if let task = taskForImage {
+                updateTask(task, withImage: image)
+                self.taskForImage = nil
+            }
+            picker.presentingViewController?.dismissViewControllerAnimated(true, completion: nil)
+    }
+    
+    // MARK: - Action
+    
+    func addAction(sender: AnyObject) {
+        Ui.showTextInputDialog(
+            onController: self,
+            withTitle: "New Task",
+            withMessage: nil,
+            withTextFieldConfig: { (textField) -> Void in
+                textField.placeholder = "Task"
+                textField.autocapitalizationType = .Sentences
+            },
+            onOk: { (task) -> Void in
+                self.createTask(task)
+            }
+        )
+    }
+    
+    // MARK: - Database
+    
+    func setupViewAndQuery() {
+        let view = database.viewNamed("tasksByCreatedAt")
+        if view.mapBlock == nil {
+            view.setMapBlock({ (doc, emit) -> Void in
+                if let type: String = doc["type"] as? String where type == "task" {
```

Comment:
You can use a single `if let` with multiple clauses, to prevent all the nesting.